### PR TITLE
YOLOv7 Assigner visualization 

### PR DESCRIPTION
**Pull request by satuoqaq**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by Nioolek — APPROVED** [review]
LGTM

**Comment by hhaAndroid** [comment]
@satuoqaq please resolve the conflicts

File: projects/assigner_visualization/README.md [hunk 1]
```diff
@@ -6,22 +6,32 @@
 
 This project is developed for easily showing assigning results. The script allows users to analyze where and how many positive samples each gt is assigned in the image.
 
-Now, the script supports `YOLOv5` and `RTMDet`.
+Now, the script supports YOLOv5, YOLOv7 and RTMDet.
```
Comment (Nioolek) [inline projects/assigner_visualization/README.md:9]:
```suggestion
Now, the script supports `YOLOv5`, `YOLOv7` and `RTMDet`.
```

File: projects/assigner_visualization/README.md [hunk 2]
```diff
@@ -6,22 +6,32 @@
...
+```
+
+RTMdet assigner visualization command:
+
 ```shell
 python projects/assigner_visualization/assigner_visualization.py projects/assigner_visualization/configs/rtmdet_s_syncbn_fast_8xb32-300e_coco_assignervisualization.py -c ${checkpont}
 ```
 
-${checkpont} is the checkpont file path. Dynamic label assignment is used in `RTMDet`, model weights will affect the positive sample allocation results, so it is recommended to load the trained model weights.
+${checkpont} is the checkpont file path. Dynamic label assignment is used in `YOLOv7` or `RTMDet`, model weights will affect the positive sample allocation results, so it is recommended to load the trained model weights.
```
Comment (Nioolek) [inline projects/assigner_visualization/README.md:35]:
```suggestion
${checkpont} is the checkpont file path. Dynamic label assignment is used in `YOLOv7` and `RTMDet`, model weights will affect the positive sample allocation results, so it is recommended to load the trained model weights.
```

File: projects/assigner_visualization/dense_heads/yolov7_head_assigner.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                    'img_inds': img_inds[retained_inds],
+                    'class_inds': class_inds[retained_inds],
+                    'retained_gt_inds': retained_gt_inds[retained_inds],
+                    'prior_ind': k
+                }
+                assign_results_feat.append(assign_results_prior)
+            assign_results.append(assign_results_feat)
+        return assign_results
+
+    def get_gt_inds(self, assigned_target, gt_instance):
```
Comment (Nioolek) [inline projects/assigner_visualization/dense_heads/yolov7_head_assigner.py:115]: add typehint and docstr

File: projects/assigner_visualization/dense_heads/yolov7_head_assigner.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            img_inds = targets_normed[:, 0]
+            class_inds = targets_normed[:, 1].long()
+            retained_gt_inds = self.get_gt_inds(
+                targets_normed, batch_targets_normed[0]).long()
+            for k in range(self.num_base_priors):
+                retained_inds = priors_inds == k
+                assign_results_prior = {
+                    'stride': self.featmap_strides[i],
+                    'grid_x_inds': grid_x_inds[retained_inds],
+                    'grid_y_inds': grid_y_inds[retained_inds],
```
Comment (Nioolek) [inline projects/assigner_visualization/dense_heads/yolov7_head_assigner.py:105]: make sure code can run on gpu

File: projects/assigner_visualization/dense_heads/yolov7_head_assigner.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                    'img_inds': img_inds[retained_inds],
+                    'class_inds': class_inds[retained_inds],
+                    'retained_gt_inds': retained_gt_inds[retained_inds],
+                    'prior_ind': k
```
Comment (Nioolek) [inline projects/assigner_visualization/dense_heads/yolov7_head_assigner.py:109]: add `offset`

File: projects/assigner_visualization/dense_heads/yolov7_head_assigner.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+            batch_targets_normed,
+            batch_img_metas[0]['batch_input_shape'],
+            self.priors_base_sizes,
+            self.grid_offset,
+            near_neighbor_thr=self.near_neighbor_thr)
+
+        mlvl_positive_infos = assigner_results['mlvl_positive_infos']
+        mlvl_targets_normed = assigner_results['mlvl_targets_normed']
+
+        assign_results = []
```
Comment (Nioolek) [inline projects/assigner_visualization/dense_heads/yolov7_head_assigner.py:65]: 加下注释：整理assign结果

**Review by Nioolek — APPROVED** [review]
LGTM

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**